compressed_image_transport/src/compressed_subscriber.cpp

```diff
@@ -50,6 +50,33 @@ namespace enc = sensor_msgs::image_encodings;
 namespace compressed_image_transport
 {
 
+void CompressedSubscriber::subscribeImpl(ros::NodeHandle& nh, const std::string& base_topic, uint32_t queue_size,
+                             const Callback& callback, const ros::VoidPtr& tracked_object,
+                             const image_transport::TransportHints& transport_hints)
+{
+    typedef image_transport::SimpleSubscriberPlugin<sensor_msgs::CompressedImage> Base;
+    Base::subscribeImpl(nh, base_topic, queue_size, callback, tracked_object, transport_hints);
+
+    // Set up reconfigure server for this topic
+    reconfigure_server_ = boost::make_shared<ReconfigureServer>(this->nh());
+    ReconfigureServer::CallbackType f = boost::bind(&CompressedSubscriber::configCb, this, _1, _2);
+    reconfigure_server_->setCallback(f);
+}
+
+
+void CompressedSubscriber::configCb(Config& config, uint32_t level)
+{
+  config_ = config;
+  if (config_.mode == "gray") {
+      flags_ = cv::IMREAD_GRAYSCALE;
+  } else if (config_.mode == "gray") {
```

Comment:
Oops, this shows I mostly tested unchanged. To test the color option, I would need a publisher explicitly sending gray images.